### PR TITLE
Add documentation for dashboards assistant plugin

### PR DESCRIPTION
**Pull request by ruanyl**
### Description
Documentation for dashboards assistant plugin.

The documentation is created by @xluo-aws @wanglam @gaobinlong @raintygao @ruanyl

### Issues Resolved
Closes https://github.com/opensearch-project/documentation-website/issues/6369

### Checklist
- [ ] By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and subject to the [Developers Certificate of Origin](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Comment by kgcreative** [comment]
This should be "OpenSearch Assistant Toolkit for OpenSearch Dashboards" (if we are specifically referring to the OpenSearch Dashboards features, as opposed to "OpenSearch Assistant Toolkit" for all the backend tooling)

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:74]:
```suggestion
### Resuming previous conversations

```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:75]:
```suggestion

```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:76]:
```suggestion
To view a previous conversation, select the clock icon to open the conversation history panel and display the chat history. A snapshot of a conversation history is shown in the following image.
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:78]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-history-entry.png" alt="Snapshot of conversation history">
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:72]:
```suggestion
<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/traces.png" alt="Snapshot of response generation details">
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:80]:
```suggestion
The conversation history can be searched by title, as shown in the following snapshot. 
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:84]:
```suggestion
#### Editing and deleting previous conversations

```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:85]:
```suggestion
Select the pencil icon to edit a conversation name and rename it. Select the **Confirm name** button to save the new name. A snapshot of the edit interface is shown in the following image.
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:87]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/edit-conversation-title.png" alt="Editing a conversation name">
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:89]:
```suggestion
Select the trash icon to delete conversation. The confirmation dialog will appear. Select “Delete conversation”. The conversation is now deleted from your chat history. A snapshot of the delete interface is shown in the following image.
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:91]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/delete-conversation.png" alt="Deleting a conversation">
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:93]:
```suggestion
### Sharing a conversation through Notebooks

```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:94]:
```suggestion
You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu on the right-side of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface. 
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:96]:
```suggestion
<img width="400px" src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/save-conversation-to-notebook.png" alt="Saving a conversation through Notebooks">
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:98]:
```suggestion
You can open your saved notebook or view a list of other notebooks by going to **Observability** > **Notebooks**. The following snapshot shows the the Notebooks interface in OpenSearch Dashboards.
```

File: _dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards-assistant/index.md:100]:
```suggestion
<img src="{{site.url}}{{site.baseurl}}/images/dashboards-assistant/conversation-in-notebook.png" alt="List of conversations saved in Notebooks">
```

**Review by vagimeli — COMMENTED** [review]
Please review the doc reviewer's edits and comments. Please make sure the edits have retained technical accuracy. Once the dev/author approves these changes, I'll format the file and replace any low-resolution screenshots. Please provide the testing environment. Following this, I'll schedule the doc for editorial review. Please be available to answer any technical questions. Thank you, Melissa

File: _dashboards-assistant/index.md
Comment (vagimeli) [reply _dashboards-assistant/index.md:100]: Add new section 

## Relate information

The following resources provide more information about the OpenSearch Assistant Toolkit and related components:

- [OpenSearch adds a new generative AI assistant toolkit](https://opensearch.org/blog/opensearch-adds-new-generative-ai-assistant-toolkit/)
- <other resources>

**Comment by vagimeli** [comment]
> This should be "OpenSearch Assistant Toolkit for OpenSearch Dashboards" (if we are specifically referring to the OpenSearch Dashboards features, as opposed to "OpenSearch Assistant Toolkit" for all the backend tooling)

@kgcreative I agree. I've changed doc title.

**Comment by vagimeli** [comment]
> @xluo-aws @wanglam @gaobinlong @raintygao @ruanyl

To move this documentation forward, please provide your technical review and approval. Once I have your approval of my rewrites and edits, I can move the content into editorial review and then merge the PR. Please reach out to me with any questions. Thank you.

File: _dashboards-assistant/index.md
Comment (vagimeli) [reply _dashboards-assistant/index.md:18]: Note: I'll add the missing link.

Comment (lezzago) [reply _dashboards-assistant/index.md:18]: Do we want to talk about the plugin in particular here? The plugin piece is part of the OpenSearch Assistant toolkit to create an chat interface for a LLM-powered OpenSearch Assistant to help users interact with OpenSearch.

**Comment by ylwu-amzn** [comment]
@ruanyl looks some screenshot should be updated

**Comment by raintygao** [comment]
> Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.

@xluo-aws  According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed, we can complete them before end of 18th, not sure if it's OK.

**Comment by raintygao** [comment]
> @ruanyl looks some screenshot should be updated

@ylwu-amzn  Could you elaborate more?

**Comment by hdhalter** [comment]
> > Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.
> 
> @xluo-aws According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed we can complete them before end of 18th, not sure if it's OK.

Hi @xluo-aws , tldr: the 18th would be too late to receive updates. Here's why: after tech review is complete (development signs off on the content from a technical perspective), we do a 'doc review' which reviews the content for clarity, format, readability, grammar, etc., and then we do an 'editorial review' which is a final check for any remaining issues (and needs a 2-day window before the release). Ideally, the content should pass the tech review no later than a week before the final date so we have time for the doc and editorial reviews. This is especially important for large PRs like this. 

Thanks to the devs that stepped in to review this!

File: _dashboards-assistant/index.md
Comment (lezzago) [inline _dashboards-assistant/index.md:42]: We should call out this is a system index, so in security enabled domains, the user must be super admin to do this.
They can use this guide to make super admin calls: https://opensearch.org/docs/latest/security/configuration/system-indices/

**Comment by vagimeli** [comment]
> > > Will update the doc by next Monday if that's ok. All related team members are taking holidays off this week.
> > 
> > 
> > @xluo-aws According to [release issue](https://github.com/opensearch-project/opensearch-build/issues/4115), pre release is on the end of February 19th. If there are any updates needed we can complete them before end of 18th, not sure if it's OK.
> 
> Hi @xluo-aws , tldr: the 18th would be too late to receive updates. Here's why: after tech review is complete (development signs off on the content from a technical perspective), we do a 'doc review' which reviews the content for clarity, format, readability, grammar, etc., and then we do an 'editorial review' which is a final check for any remaining issues (and needs a 2-day window before the release). Ideally, the content should pass the tech review no later than a week before the final date so we have time for the doc and editorial reviews. This is especially important for large PRs like this.
> 
> Thanks to the devs that stepped in to review this!

@hdhalter @xluo-aws I've moved this into editorial review. I'll address editor comments when they come in. If any content is added after the editorial review, I'll review them and ask the editor to review only the newly added sections. This will help keep us on track for publishing by release date.

File: _dashboards-assistant/index.md
Comment (natebower) [inline _dashboards-assistant/index.md]:
```suggestion
You can use [Notebooks]({{site.url}}{{site.baseurl}}/observing-your-data/notebooks/) to save your conversations. To use this option, select `Save to notebook` from the dropdown menu to the right of **OpenSearch Assistant**. Enter a name for the notebook and save it. All conversations (questions and responses) between you and the large language model (LLM) will be saved to this notebook. The following snapshot shows a save notebook interface. 
```

Comment (natebower) [inline _dashboards-assistant/index.md]: "screenshot" instead of "snapshot"?

File: _dashboards-assistant/index.md
Comment (natebower) [inline _dashboards-assistant/index.md]:
```suggestion
You can open your saved notebook or view a list of other notebooks by going to **Observability** > **Notebooks**. The following snapshot shows the Notebooks interface in OpenSearch Dashboards.
```

Comment (natebower) [inline _dashboards-assistant/index.md]: "screenshot" instead of "snapshot"?

**Review by natebower — COMMENTED** [review]
@vagimeli @ruanyl Please see my comments and changes and let me know if you have any questions. Thanks!

File: _dashboards-assistant/index.md
Comment (vagimeli) [reply _dashboards-assistant/index.md]: Yes, I'll make changes throughout.

**Comment by vagimeli** [comment]
> @ruanyl looks some screenshot should be updated

@ruanyl If you send me the test environment, I can update all screenshots.

File: _dashboards-assistant/index.md
Comment (vagimeli) [reply _dashboards-assistant/index.md:44]: @lezzago Please review the following callout text to make sure I've captured your comment accurately:

_This example is a system index. In security-enabled domains, only super administrators have access to execute this code. For details about making super admin calls, see the [System indexes](https://opensearch.org/docs/latest/security/configuration/system-indices/) guide. For access permission, contact your IT administrator._

**Comment by vagimeli** [comment]
> @vagimeli @ruanyl Please see my comments and changes and let me know if you have any questions. Thanks!

@natebower @ruanyl I've accepted the editorial review comments.

**Comment by ylwu-amzn** [comment]
> > @ruanyl looks some screenshot should be updated
> 
> @ylwu-amzn Could you elaborate more?

Shared you a latest test cluster. You can take new screenshots there.

**Comment by vagimeli** [comment]
> > > @ruanyl looks some screenshot should be updated
> > 
> > 
> > @ylwu-amzn Could you elaborate more?
> 
> Shared you a latest test cluster. You can take new screenshots there.

@ylwu-amzn @ruanyl I've replaced screenshots based on the the test environment you shared. Please provide your approval of this PR if its ready for merging. Thank you, Melissa

File: _dashboards/dashboards-assistant/index.md
Comment (vagimeli) [inline _dashboards/dashboards-assistant/index.md:43]: @natebower Please edit lines 43-44.

**Review by vagimeli — APPROVED** [review]
Doc and editorial reviews completed. Editorial feedback addressed.

**Pull request merged**
